### ISSUE TITLE
cannot reproduce the numbers in table 1 on the sum dataset

### ISSUE THREAD

**Issue by husha1993** [FerranAlet/modular-metalearning#3]
Hi,

I tried to reproduce the numbers in table 1 on the sum dataset, while I found the MAML outperforms the BounceGrad. 
BounceGrad is about 12.96 almost the same as in table1 while MAML is about 4.25.
I used the command as follows to reproduce MAML on the sum dataset:
python modular_main.py --type_modules affine-1-64-64-1 --num_modules 1 --composer sum-1 --meta_lr 0.003 --plot_name MAML --limit_data 80 --optimization_steps 3000 --split_by_file --meta_split 90,10,0 --data_split 20,80,0 --data_desc HDF5@sums.hdf5 --meta_batch_size 32 --max_datasets 300 --MAML --MAML_step_size 0.1
Is there anything wrong with the command?
Thanks.

**Comment by FerranAlet** [comment]
Ok, as we talked via email:
- Both lines are running the version with 16 pts and thus we should get results of about 8.0 for MAML and 0.4 for BounceGrad.
- One issue was looking at a graph that was not meta-validation validation (i.e. meta-test test); MAML tends to overfit to both meta-training and meta-validation training data, thus getting lower scores on those graphs. This was the reproduced graph, with MAML converging going down to 10 then a bit up, and BounceGrad around 14, but clearly going down.
![old_scheduler](https://user-images.githubusercontent.com/9846005/72301514-f97adf00-3634-11ea-8ea7-caeb9ada9c1e.png)

As you suggested in the email chain, the slowness of BounceGrad could be due to a small change in [pytorch lr scheduler](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate/). In any case, making it a bit more patient ([just pushed](https://github.com/FerranAlet/modular-metalearning/commit/783ebc40ecfe0bb7fd98b63ddd8bf9ebb28a1b5c)) gets us back to results very similar to the reported ones (MAML with new scheduler in grey going down to 0.08/0.09, BounceGrad with new scheduler in orange going down to 0.01).
![moma-current-lr](https://user-images.githubusercontent.com/9846005/72301965-85413b00-3636-11ea-86f3-01e397b8cd8f.png)

Finally, as mentioned in the paper, since there are no inner gradients BounceGrad is faster per step, so the same graph looks like this in terms of training time:
![moma-current-lr-time](https://user-images.githubusercontent.com/9846005/72302044-c89ba980-3636-11ea-9175-74f5ce398b30.png)
